L1Trigger/L1THGCal/test/HGCalBackendStage1ParameterExtractor.cc

```diff
@@ -0,0 +1,293 @@
+  triggerGeometry_ = es.getHandle(triggerGeomToken_);
+
+  json outJSON;
+
+  // fill MetaData
+  outJSON["MetaData"]["CMSSWversion"] = cmssw_version_;
+  outJSON["MetaData"]["fpgaId"] = the_fpga_;
+
+  // fill geometry configuration
+  outJSON["TriggerGeometryConfig"]["TriggerGeometryName"] = trigger_geom_;
+  outJSON["TriggerGeometryConfig"]["DisconnectedLayers"] = disconnected_layers_;
+  outJSON["TriggerGeometryConfig"]["DisconnectedModules"] = disconnected_modules_;
+  outJSON["TriggerGeometryConfig"]["JsonMappingFile"] = json_mapping_file_;
+  outJSON["TriggerGeometryConfig"]["L1TLinksMapping"] = L1T_links_mapping_;
+  outJSON["TriggerGeometryConfig"]["L1TModulesMapping"] = L1T_modules_mapping_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorLinksPerModule"] = scintillator_links_per_module_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorModuleSize"] = scintillator_module_size_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorTriggerCellSize"] = scintillator_trigger_cell_size_;
+
+  // fill truncation parameters
+  outJSON["TruncationConfig"]["rozMin"] = roz_min_;
+  outJSON["TruncationConfig"]["rozMax"] = roz_max_;
+  outJSON["TruncationConfig"]["rozBins"] = roz_bins_;
+  outJSON["TruncationConfig"]["maxTcsPerBin"] = max_tcs_per_bins_;
+  outJSON["TruncationConfig"]["phiSectorEdges"] = phi_edges_;
+
+  // fill trigger geometry
+  fillTriggerGeometry(outJSON);
+
+  // write out JSON file
+  std::ofstream outputfile(outJSONname_.c_str());
+  outputfile << std::setw(4) << outJSON << std::endl;
+}
+
+void HGCalBackendStage1ParameterExtractor::fillTriggerGeometry(json& json_file) {
+  trigger_map_set trigger_cells;
+
+  // retrieve valid trigger cells
+  std::cout << "Getting EE trigger cells\n";
+  for (const auto& id : triggerGeometry_->eeGeometry()->getValidDetIds()) {
+    HGCSiliconDetId detid(id);
+    if (!triggerGeometry_->eeTopology().valid(id))
+      continue;
+    uint32_t trigger_cell = triggerGeometry_->getTriggerCellFromCell(id);
+    auto itr_insert = trigger_cells.emplace(trigger_cell, std::unordered_set<uint32_t>());
+    itr_insert.first->second.emplace(id);
+  }
+  std::cout << "Getting HSi trigger cells\n";
+  for (const auto& id : triggerGeometry_->hsiGeometry()->getValidDetIds()) {
+    HGCSiliconDetId detid(id);
+    if (!triggerGeometry_->hsiTopology().valid(id))
+      continue;
+    uint32_t trigger_cell = triggerGeometry_->getTriggerCellFromCell(id);
+    auto itr_insert = trigger_cells.emplace(trigger_cell, std::unordered_set<uint32_t>());
+    itr_insert.first->second.emplace(id);
+  }
+  std::cout << "Getting HSc trigger cells\n";
+  for (const auto& id : triggerGeometry_->hscGeometry()->getValidDetIds()) {
+    uint32_t trigger_cell = triggerGeometry_->getTriggerCellFromCell(id);
+    auto itr_insert = trigger_cells.emplace(trigger_cell, std::unordered_set<uint32_t>());
+    itr_insert.first->second.emplace(id);
+  }
```

Comment:
Storing the cells within the trigger cells is in principle not needed. Only the list of TCs is needed.

Reply:
I cleaned that part out and fill a vector with `uint32_t` IDs instead to get the TCs